services/avena_fm/scripts/avena_fm_demo_epy_block_0.py

```diff
@@ -0,0 +1,46 @@
+import os
+import numpy as np
+from gnuradio import gr
+from pynats2 import NATSClient
+import json
+from base64 import b64encode, b64decode
+
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return json.JSONEncoder.default(self, obj)
+
+class blk(gr.sync_block):  # other base classes are basic_block, decim_block, interp_block
+
+
+    def __init__(self, vector_size=1024, subject='fft', 
+                 nats_server='nats://localhost:4222',
+                 freq=98700000, span=2000000):  # only default arguments here
+
+        gr.sync_block.__init__(
+            self,
+            name='NATS Sink',   # will show up in GRC
+            in_sig=[(np.ushort, vector_size)],
+            out_sig=None
+        )
+
+        #hostname = os.getenv('HOSTNAME')
+        self.vector_size = vector_size
+        self.freq = freq
+        self.span = span
+        self.subject = os.getenv('HOSTNAME') + '.' + subject
+        self.nc = NATSClient(nats_server, socket_timeout=2)
+        self.nc.connect()
+        
+    def work(self, input_items, output_items):
+        
+        b64encpayload = str(b64encode(input_items[0][0]), 'utf-8')
+        json_dump = json.dumps({'fft': b64encpayload,
+                                'fc' : self.freq, 
+                                'span' : self.span, 
```

Comment:
I think I'll need gain to compute the FFt y-axis and set the slider properly

Reply:
I suppose I'll need stream too.

That or we need a req/reply for me to ask for the current value of everything.

Seems easier for now to just send it all every time

Reply:
Both audio and FFT streams stop when stream is False.

Reply:
I see. 

This has some problems still too, like if I close the app with it on then it will just stream and use data anyway.